src/components/Notification.tsx

```diff
@@ -0,0 +1,186 @@
+      actorName,
+      countOtherActors,
+    });
+  }
+  if (verb === 'comment') {
+    return t('{{ actorName }} and {{ countOtherActors }} others commented on your {{ activityVerb }}', {
+      actorName,
+      activityVerb,
+      countOtherActors,
+    });
+  }
+
+  console.warn('No notification styling found for your verb, please create your own custom Notification group.');
+  return '';
+};
+
+export const Notification = <
+  UT extends DefaultUT = DefaultUT,
+  AT extends DefaultAT = DefaultAT,
+  CT extends UR = UR,
+  RT extends UR = UR,
+  CRT extends UR = UR
+>({
+  activityGroup,
+  onMarkAsRead,
+  onClickUser,
+  onClickNotification,
+}: NotificationProps<UT, AT, CT, RT, CRT>) => {
+  const { t, tDateTimeParser } = useTranslationContext();
+  const activities = activityGroup.activities;
+  const latestActivity = activities[0];
+
+  if (typeof latestActivity.object === 'string') return null;
+
+  const lastObject = latestActivity.object as EnrichedActivity;
+  const lastActor = userOrDefault(latestActivity.actor);
+  const headerText = getHeaderText(t, activities.length, latestActivity.verb, lastActor.data.name, lastObject.verb);
+  const handleUserClick = useOnClickUser<UT>(onClickUser);
+  const handleNotificationClick = onClickNotification
+    ? (e: SyntheticEvent) => {
+        e.stopPropagation();
+        onClickNotification(activityGroup);
+      }
+    : undefined;
+
+  return (
+    <div
+      onClick={handleNotificationClick}
+      className={'raf-notification' + (activityGroup.is_read ? ' raf-notification--read' : '')}
+    >
+      <Avatar
+        onClick={handleUserClick?.(lastActor as EnrichedUser<UT>)}
+        image={lastActor.data.profileImage}
+        circle
+        size={30}
+      />
+
+      <div className="raf-notification__content">
+        <div className="raf-notification__header">
+          <strong>{headerText}</strong>
+          {!activityGroup.is_read && onMarkAsRead && (
+            <Dropdown>
+              <div>
+                <Link
+                  onClick={(e) => {
+                    e.stopPropagation();
+                    onMarkAsRead(activityGroup);
+                  }}
+                >
+                  Mark&nbsp;as&nbsp;read
+                </Link>
+              </div>
+            </Dropdown>
+          )}
+        </div>
+        <div>
+          <small>{humanizeTimestamp(latestActivity.time, tDateTimeParser)}</small>
+        </div>
+        {latestActivity.verb !== 'follow' && (
+          <AttachedActivity activity={latestActivity.object as EnrichedActivity<UT, AT, CT, RT, CRT>} />
+        )}
+      </div>
+
+      <div className="raf-notification__extra">
+        {activities.length > 1 && latestActivity.verb === 'follow' && (
+          <AvatarGroup
+            onClickUser={onClickUser}
+            avatarSize={30}
+            users={getUsers<UT, AT, CT, RT, CRT>(activities.slice(1, activities.length)) as Array<EnrichedUser<UT>>}
```

Comment:
```suggestion
            users={getUsers<UT, AT, CT, RT, CRT>(rest)>}
```
Could probably be memoized too.